File: examples/workflows/01-simple-sequential/Main.java
/*
 * Example 1: Simple Sequential Workflow - Java
 *
 * This example shows the most basic AxonFlow workflow: send a query to an LLM and get a response.
 */
package com.getaxonflow.examples;

import com.getaxonflow.sdk.AxonFlow;
import com.getaxonflow.sdk.AxonFlowConfig;
import com.getaxonflow.sdk.types.ExecuteQueryRequest;
import com.getaxonflow.sdk.types.ExecuteResponse;

import java.util.Map;
import java.util.Optional;

public class Main {

    public static void main(String[] args) {
        // Get AxonFlow configuration from environment
        String agentUrl = Optional.ofNullable(System.getenv("AXONFLOW_AGENT_URL"))
                .orElse("http://localhost:8080");
        String clientId = System.getenv("AXONFLOW_CLIENT_ID");
        String clientSecret = System.getenv("AXONFLOW_CLIENT_SECRET");

        if (clientId == null || clientId.isEmpty() || clientSecret == null || clientSecret.isEmpty()) {
            System.err.println("❌ AXONFLOW_CLIENT_ID and AXONFLOW_CLIENT_SECRET must be set");
            System.exit(1);
        }

        // Create AxonFlow client
        AxonFlowConfig config = AxonFlowConfig.builder()
                .endpoint(agentUrl)
                .clientId(clientId)
                .clientSecret(clientSecret)
                .build();

        AxonFlow client = new AxonFlow(config);

        System.out.println("✅ Connected to AxonFlow");

        // Define a simple query
        String query = "What is the capital of France?";
        System.out.println("📤 Sending query: " + query);

        try {
            // Send query to AxonFlow
            ExecuteResponse response = client.proxyLLMCall(
                    ExecuteQueryRequest.builder()
                            .userToken("user-123")
                            .query(query)
                            .requestType("chat")
                            .context(Map.of("model", "gpt-4"))
                            .build()
            );

            // Print response
            System.out.println("📥 Response: " + response.getData());
            System.out.println("✅ Workflow completed successfully");
        } catch (Exception e) {
            System.err.println("❌ Query failed: " + e.getMessage());
            System.exit(1);
        }
    }
}

File: examples/workflows/01-simple-sequential/main.go
// Simple Sequential Workflow Example
//
// VALIDATION: This example exits with code 1 if any assertion fails.
//
// Demonstrates a simple sequential LLM call through AxonFlow governance.
package main

import (
	"fmt"
	"os"

	axonflow "github.com/getaxonflow/axonflow-sdk-go/v9"
)

var failures []string

func assertCheck(condition bool, message string) {
	if condition {
		fmt.Printf("   ✓ PASS: %s\n", message)
	} else {
		fmt.Printf("   ❌ FAIL: %s\n", message)
		failures = append(failures, message)
	}
}

func main() {
	fmt.Println("Simple Sequential Workflow - Go")
	fmt.Println("================================")
	fmt.Println()

	// Create AxonFlow client (no auth required for community mode)
	client := axonflow.NewClient(axonflow.AxonFlowConfig{
		Endpoint: getEnv("AXONFLOW_ENDPOINT", "http://localhost:8080"),
	})

	// Test 1: Health check
	fmt.Println("Test 1: Health Check")
	fmt.Println("--------------------")
	err := client.HealthCheck()
	assertCheck(err == nil, "Agent is healthy")
	if err != nil {
		fmt.Printf("   Error: %v\n", err)
	}
	fmt.Println()

	// Test 2: Send a simple query
	fmt.Println("Test 2: Simple Query")
	fmt.Println("--------------------")
	query := "What is the capital of France?"
	fmt.Printf("   Query: %s\n", query)

	response, err := client.ProxyLLMCall("", query, "chat", nil)
	assertCheck(err == nil, "LLM call does not return error")
	if err == nil {
		assertCheck(response.Success, "LLM response is successful")
		assertCheck(response.Data != nil, "Response has data")
		fmt.Printf("   Response: %v\n", truncate(fmt.Sprintf("%v", response.Data), 80))
	} else {
		fmt.Printf("   Error: %v\n", err)
	}
	fmt.Println()

	// Summary
	fmt.Println("================================")
	if len(failures) == 0 {
		fmt.Println("✓ ALL TESTS PASSED")
		os.Exit(0)
	} else {
		fmt.Printf("❌ %d TEST(S) FAILED:\n", len(failures))
		for _, f := range failures {
			fmt.Printf("   - %s\n", f)
		}
		os.Exit(1)
	}
}

func getEnv(key, defaultValue string) string {
	if value := os.Getenv(key); value != "" {
		return value
	}
	return defaultValue
}

func truncate(s string, maxLen int) string {
	if len(s) > maxLen {
		return s[:maxLen] + "..."
	}
	return s
}
